### PR TITLE
Add option for updating to the latest GitHub release

### PR DESCRIPTION
**Pull request by bellerbrock**
## Description

Adds a 'Check for Updates' tool to the About screen. Checks for more recent Github releases. If any are found gives the option to download and update to the latest release.  

Closes # #567

## Type of change

_What type of changes does your code introduce? Put an `x` in boxes that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated relevant documentation

## Changelog entry

_Please add a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

Added a 'Check for Updates' option to the About section

```

### CODE REVIEW

**Comment by chaneylc** [comment]
This looks really cool, it all works up until I press install I get a system dialog that says: "App not installed as package conflicts with existing package."

File: app/src/main/java/com/fieldbook/tracker/utilities/VersionChecker.java
Comment (chaneylc) [inline app/src/main/java/com/fieldbook/tracker/utilities/VersionChecker.java:1]: AsyncTask is deprecated so we should be using kotlin coroutines or work manager, not imperative for this PR to be approved.

**Review by chaneylc — CHANGES_REQUESTED** [review]
Looks good but it's not working. FB says I have 5.4.6.11 version, when I check for a new version I get 5.4.8.13. Then it shows the message I described before.

**Comment by chaneylc** [comment]
Oops I wasn't using the pre built apk! It works with that one :)

**Review by chaneylc — APPROVED** [review]

**Comment by bellerbrock** [comment]
@chaneylc thanks for your review. And your additions to improve the context handling and string management!

I got the same error message at first when testing in an emulator. Had to use the (signed) pre built apk, and test on my physical device.

Will have to follow up with you later on [this part](https://github.com/PhenoApps/Field-Book/blob/3eee31cafcf8c66081d4578b8dbb0ef9479d5076/app/src/main/java/com/fieldbook/tracker/utilities/VersionChecker.java#L45), once I've advanced a bit more with Kotlin.

**Pull request merged**